sshd-core/src/main/java/org/apache/sshd/client/kex/DHGClient.java

```diff
@@ -124,10 +129,67 @@ public boolean next(int cmd, Buffer buffer) throws Exception {
 
         buffer = new ByteArrayBuffer(k_s);
         serverKey = buffer.getRawPublicKey();
-        String keyAlg = KeyUtils.getKeyType(serverKey);
+        PublicKey serverPublicHostKey = serverKey;
+
+        OpenSshCertificate openSshKey = null;
```

Comment:
Why is it defined outside the `if` block ? I don't see it being used outside it...